src/Java.Interop/Java.Interop/JniRuntime.JniTypeManager.cs

```diff
@@ -226,8 +226,8 @@ protected bool TryRegisterNativeMembers (JniType nativeClass, Type type, string
 
 			static bool TryLoadJniMarshalMethods (JniType nativeClass, Type type, string methods)
 			{
-				var marshalType = type.GetTypeInfo ()?.GetDeclaredNestedType ("__<$>_jni_marshal_methods")?.AsType ();
-				if (marshalType == null || !(marshalType is Type))
+				var marshalType = type?.GetNestedType ("__<$>_jni_marshal_methods", BindingFlags.Public | BindingFlags.NonPublic);
```

Comment:
Will these nested types be public or private? I could drop one flag here.

Reply:
Currently it's emitted as `TypeAttributes.NestedPrivate`:

https://github.com/xamarin/java.interop/blob/4565369743a58a1d43e5a79c406c27d46b81bb08/tools/jnimarshalmethod-gen/TypeMover.cs#L91

You should be able to remove the `BindingFlags.Public`.